src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4153,16 +4153,39 @@ public override bool PreVisit(CallNode node)
                     return false;
                 }
 
+                var numOverloads = overloads.Count();
+
+                var overloadsWithUntypedObjectLambdas = overloadsWithLambdas.Where(func => func.HasUntypedObjectLambdas);
+                TexlFunction overloadWithUntypedObjectLambda = null;
+                if (overloadsWithUntypedObjectLambdas.Any())
+                {
+                    Contracts.Assert(overloadsWithUntypedObjectLambdas.Count() == 1, "Incorrect multiple overloads with UntypedObject lambdas.");
+                    overloadWithUntypedObjectLambda = overloadsWithUntypedObjectLambdas.Single();
+                    Contracts.Assert(overloadWithUntypedObjectLambda.HasLambdas);
+
+                    // As an extraordinarily spcial case, we ignore untype object lambdas for now, and type check as normal
+                    // using the function without untyped object params. This only works if both functions have exactly
+                    // the same arity (this is enforced below).
```

Comment:
Can we try to differentiate the overloads based on their parameters which *aren't* lambdas? In the ForAll case, the type of the first argument is well-defined, so we should be able to resolve it without using this workaround

Reply:
Unfortunately, the user can make an incorrect call without arguments. ForAll() would not be able to be resolved. Note that arity checks happen after we actually have an overload.